hibernate-core/src/main/java/org/hibernate/sql/results/graph/embeddable/internal/EmbeddableFetchImpl.java

```diff
@@ -175,7 +176,8 @@ public EmbeddableInitializer<?> createInitializer(InitializerParent<?> parent, A
 
 	@Override
 	public boolean appliesTo(GraphImplementor<?> graphImplementor, JpaMetamodel metamodel) {
-		return getFetchParent().appliesTo( graphImplementor, metamodel );
+		return GraphHelper.appliesTo( graphImplementor, metamodel.embeddable( getResultJavaType().getTypeName() ) )
+				|| getFetchParent().appliesTo( graphImplementor, metamodel );
```

Comment:
It seems to me that the call to `getFetchParent().appliesTo()` is wrong. The passed `graphImplementor` must AFAIU refer to a type compatible to the mapping type.

```suggestion
		return GraphHelper.appliesTo( graphImplementor, metamodel.embeddable( getResultJavaType().getTypeName() ) );
```

Reply:
I also thought of this, and initially I made this change as well. A test was failing though, as there is a case where an entity graph is applied to fetch the properties of an `@IdClass` embeddable fetch instance - that's why the previous implementation checked `getFetchParent().appliesTo()` I think.

I've now changed it to `metamodel.managedType()`: this should handle both "normal" embeddables and this special case where `getResultJavaType().getTypeName()` corresponds to the entity containing the non-aggregated identifier, and added a comment to clarify that.